pkg/apis/security/types_test.go

```diff
@@ -0,0 +1,87 @@
+package security_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/elasticsearch-proxy/pkg/apis/security"
+	cl "github.com/openshift/elasticsearch-proxy/pkg/handlers/clusterlogging/types"
+	test "github.com/openshift/elasticsearch-proxy/test"
+)
+
+func newProjects(projects ...string) []cl.Project {
+	result := []cl.Project{}
+	for _, project := range projects {
+		result = append(result, cl.Project{project, "123abc"})
+	}
+	return result
+}
+
+var _ = Describe("Generating Security roles", func() {
+	var (
+		docs   security.ACLDocuments
+		users  []cl.UserInfo
+		result string
+		err    error
+	)
+	BeforeEach(func() {
+		docs = security.ACLDocuments{}
+		docs.Set(security.NewRoles())
+		docs.Set(security.NewRolesMapping())
+		users = []cl.UserInfo{
+			{
+				Username: "user2.bar@email.com",
+				Groups:   []string{},
+				Projects: newProjects("xyz", "foo.bar"),
+			},
+			{
+				Username: "CN=jdoe,OU=DL IT,OU=User Accounts,DC=example,DC=com",
+				Groups:   []string{"myspecialgroup"},
+				Projects: newProjects("distinguishedproj"),
+			},
+		}
+		for _, user := range users {
+			docs.AddUser(&user, 15)
+		}
+	})
+	Describe("for shared_ops kibana index mode", func() {
+
+		It("should produce a well formed roles.yaml", func() {
+			result, err = docs.Roles().ToYaml()
+			Expect(err).To(BeNil())
+			test.Expect(result).ToMatchYaml(`
```

Comment:
This will work as long as the order of elements is always preserved going to and from yaml.  I'm assuming it works that way now, and this test always passes.  Otherwise, we may have to convert the expected yaml string to a `Roles` and compare to `docs.Roles()`